[FIX] base: add `commercial_company_name` to the depends

pedrambiria opened the pull request:
Before this commit: if you add a custom one2many field to the 'res.partner', like x_related_commercial_partner_ids, that is related to the `commercial_partner_id` in the`res.partner` model, it won't update the display name of a contact in case of changing its parent_id name.

Here are the steps to reproduce the problem:
 1. Create a new custom field with these values:
   a. Field Type = one2many
   b. Model = Contact
   c. Related Model = res.partner
   d. Relation Field = commercial_partner_id
 2. Create a new Contact that is the "Company" (e.g. "My Company")
 3. Create a new Contact that is the "Individual" (e.g. "My Name"), and
    put the "My Company" as its parent_id.
 4. Now the display_name is "My Company, My Name" which is correct
 5. Change the company name to "My new Company"
  -> display_name won't change, and is "My Company, My Name"

The solution is to add the 'commercial_company_name' to the `display_name` depends.

opw-3202894

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/114344).

pedrambiria commented:
@xmo-odoo Thanks for your feedback.
I updated my commit message to include clear steps for reproducing the issue.
I don't know exactly why it causes this problem, but I know that when it tries to compute the `display_name` inside the `_compute_display_name` function, it will read the `commercial_company_name` from the cache and won't recompute it. But after the `_compute_display_name` execution, the `commercial_company_name` compute function will be called to recompute it. So, as the `commercial_company_name` is using the compute the `display_name`, I added it into its depends.

ryv-odoo commented:
Hello @pedrambiria 

Thank you for your work. LTGM.
@xmo-odoo , Adding a one2many can change the behavior of `_modified_triggers`, [here](https://github.com/odoo/odoo/blob/ca2637671b409978f31ddf1146b3567699062d08/odoo/models.py#L5863-L5885). For this case, I didn't check in details, but the fix seems legit and logical.

@robodoo r+
